AzFilesHybrid/AzFilesHybrid.psm1

```diff
@@ -4006,11 +4004,118 @@ function Debug-AzStorageAccountEntraKerbAuth {
     }
 }
 
+function Debug-RBACCheck {
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory=$True, Position=0, HelpMessage="Storage account name")]
+        [string]$StorageAccountName,
+        [Parameter(Mandatory=$True, Position=1, HelpMessage="User Principal name")]
+        [string]$UserPrincipalName,
+        [Parameter(Mandatory=$True, Position=2, HelpMessage="Check result object")]
+        [CheckResult]$checkResult
+    )
+    process {
+        try {
+            Request-ConnectMsGraph `
+                    -Scopes "User.Read.All", "GroupMember.Read.All" `
+                    -RequiredModules @("Microsoft.Graph.Users", "Microsoft.Graph.Groups", "Microsoft.Graph.Identity.DirectoryManagement")
+                    
+            $userOid = $(Get-MgUser -Filter "UserPrincipalName eq $UserPrincipalName" -Property Id).Id
+
+            $groupIds = Get-MgUserMemberOf -UserId $userOid | Select-Id
+
+            $roleNames = @(
+                "Storage File Data SMB Share Reader",
+                "Storage File Data SMB Share Contributor",
+                "Storage File Data SMB Share Elevated Contributor"
+            )
+
+            $listOfRoleNames = @{}
+            foreach ($roleName in $roleNames) 
+            {
+                $scope = "/subscriptions/$subscription/resourceGroups/$resourceGroup/providers/Microsoft.Storage/storageAccounts/$storageAccount/fileServices/default/fileshares/$fileShare"
+                $assignments = Get-AzRoleAssignment -RoleDefinitionName $roleName -Scope $scope
+                
+                foreach ($assignment in $assignments) 
+                {
+                    if ($assignment.ObjectType -eq "User") 
+                    {
+                        if ($assignment.ObjectId -eq $userOid) 
+                        {
+                            $listOfRoleNames.Add($roleName)
+                            break
+                        }
+                    }
+                    elseif ($assignment.ObjectType -eq "Group") 
+                    {
+                        if ($groupIds -contains $assignment.ObjectId) 
+                        {
+                            $listOfRoleNames.Add($roleName)
+                            break
+                        }
+                    }
+                }
+            }
+
+            if ($listOfRoleNames.size -eq 0) {
+                $message = "User '$($user.UserPrincipalName)' is not assigned any SMB share-level permission to" `
+                        + " storage account '$StorageAccountName' in resource group '$ResourceGroupName'. Please" `
+                        + " configure proper share-level permission following the guidance at" `
+                        + " https://docs.microsoft.com/en-us/azure/storage/files/storage-files-identity-ad-ds-assign-permissions"
+                    Write-Error -Message $message -ErrorAction Stop
+            }

```

Comment:
Same here, let's ad an `else` that logs the `$listOfRoleNames` and explains how your RBAC is given.

Reply:
Didnt understand this else part

Reply:
Looks good now :)